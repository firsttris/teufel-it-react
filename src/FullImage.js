import React, { Component } from 'react';

class FullImage extends Component {
  render() {
    let imageStyle = {
      position: 'relative',
      opacity: '0.65',
      backgroundAttachment: 'fixed',
      backgroundPosition: 'center',
      backgroundRepeat: 'no-repeat',
      backgroundImage: `url(${this.props.image})`,
      height: this.props.height, 
      backgroundSize: 'cover',
      minHeight: '400px'
    }

    return (
      <div style={imageStyle}>
      </div>
    );
  }
}

export default FullImage;
