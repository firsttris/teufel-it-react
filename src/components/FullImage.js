import React, {Component} from 'react';
import './fullImage.css';
import PropTypes from 'prop-types';

class FullImage extends Component {
    render() {
        let imageStyle = {
            position: 'relative',
            opacity: '0.65',
            backgroundAttachment: 'fixed',
            backgroundPosition: 'center',
            backgroundRepeat: 'no-repeat',
            backgroundImage: `url(${this.props.image})`,
            height: this.props.height,
            backgroundSize: 'cover',
            minHeight: '400px'
        };

        return (
            <div className="row">
                <div className="col" style={{padding: 0}}>
                    <div className="full-image" style={imageStyle}>
                    </div>
                </div>
            </div>
        );
    }
}

FullImage.propTypes = {
    image: PropTypes.string,
    height: PropTypes.string
};

export default FullImage;