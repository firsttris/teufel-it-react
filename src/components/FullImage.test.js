import React from 'react';
import FullImage from './FullImage';
import renderer from 'react-test-renderer';
import Paper from './../assets/images/backgrounds/paper.png';

test('Display FullImage', () => {
  const component = renderer.create(
    <FullImage image={Paper} height="550px"></FullImage>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});