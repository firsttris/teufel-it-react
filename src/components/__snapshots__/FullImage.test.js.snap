// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display FullImage 1`] = `
<div
  style={
    Object {
      "backgroundAttachment": "fixed",
      "backgroundImage": "url(paper.png)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "height": "550px",
      "minHeight": "400px",
      "opacity": "0.65",
      "position": "relative",
    }
  }
/>
`;
